include/libvfio-user.h

```diff
@@ -501,7 +501,13 @@ typedef struct {
     /*
      * Migration state transition callback.
      *
-     * The callback should return -1 on error, setting errno.
+     * The callback should return 0 on success and -errno on error.
+     *
+     * When operating in non-blocking mode (LIBVFIO_USER_FLAG_ATTACH_NB was
```

Comment:
is there a specific reason to require this?

Reply:
Purely for simplicity since it's only SPDK that requires it and it doesn't use non-blocking mode. I haven't thought of the implications in allowing this in blocking mode and since we don't need it now I think it's best to postpone implementing this.

Reply:
sure

Reply:
I'll drop a comment for future reference.